WordPress/src/main/java/org/wordpress/android/viewmodel/activitylog/ActivityLogViewModel.kt

```diff
@@ -554,6 +618,56 @@ class ActivityLogViewModel @Inject constructor(
         }
     }
 
+    fun onQueryBackupDownloadStatus(rewindId: String, downloadId: Long) {
+        queryBackupDownloadStatus(downloadId)
+        showBackupDownloadStartedMessage(rewindId)
+    }
+
+    private fun queryBackupDownloadStatus(downloadId: Long? = null) {
+        backupDownloadStatusJob?.cancel()
+        backupDownloadStatusJob = viewModelScope.launch {
+            getBackupDownloadStatusUseCase.getBackupDownloadStatus(site, downloadId)
+                    .collect { state -> handleBackupDownloadStatus(state) }
+        }
+    }
+
+    private fun handleBackupDownloadStatus(state: BackupDownloadRequestState) {
+        when (state) {
+            is BackupDownloadRequestState.Progress -> if (!isBackupDownloadProgressItemShown) {
+                reloadEvents(
+                        backupDownloadEvent = BackupDownloadEvent(
+                                displayProgress = true,
+                                isCompleted = false,
+                                rewindId = state.rewindId,
+                                published = state.published
+                        )
+                )
+            }
+            is BackupDownloadRequestState.Complete -> if (isBackupDownloadProgressItemShown) {
```

Comment:
Nitpick 🔍: In a when statement, if a branch is more than a single line, consider separating it from adjacent case blocks with a blank line.

Feel free to ignore it if you preferred consistency over this.

Reply:
👋 @ashiagr !

Thanks for the comment. 🙏 Can you please elaborate on that because I am not 100% sure I understand?

Also, what if I do the below and extract the logic in private fucntions:
```
    private fun handleBackupDownloadStatus(state: BackupDownloadRequestState) {
        when (state) {
            is BackupDownloadRequestState.Progress -> handleBackupDownloadStatusForProgress(state)
            is BackupDownloadRequestState.Complete -> handleBackupDownloadStatusForComplete(state)
            else -> Unit // Do nothing
        }
    }
```

Reply:
PS: Same apply for the `Restore` status:
```
    private fun handleRestoreStatus(state: RestoreRequestState) {
        when (state) {
            is RestoreRequestState.Progress -> handleRestoreStatusForProgress(state)
            is RestoreRequestState.Complete -> handleRestoreStatusForComplete(state)
            else -> Unit // Do nothing
        }
    }
```

Reply:
It's such a minor thing, I just wanted to know what you think of this example from [Formatting control flow statements](https://kotlinlang.org/docs/reference/coding-conventions.html#formatting-control-flow-statements) 

```
private fun parsePropertyValue(propName: String, token: Token) {
    when (token) {
        is Token.ValueToken ->
            callback.visitValue(propName, token.value)
<blank line>
        Token.LBRACE -> { // ...
        }
    }
}
```

> what if I do the below and extract the logic in private fucntions:

Works for me 👍

Reply:
Hmm... Didn't know that this convention existed @ashiagr , thanks for letting me know! 🙏 

I think what that convention tries to say applies when you don't have braces, then it does make sense to have an empty line in between in order to avoid confusion. 👍 However, we you do have braces, maybe this does not apply since the end brace is already adding this effect, wdyt?

PS: I make the extract refactoring for now, but will definitely keep that in mind going forward, thanks for sharing! ❤️

Reply:
This is now done here: b0c8190125bd76c4d8c2d20d5bb382a06f4fd945

Reply:
Thank you for the explanation, makes sense 🙇‍♀️.